frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -150,10 +182,15 @@ export const funnelLogic = kea<funnelLogicType<TimeStepOption>>({
                     }
                     breakpoint()
                     insightLogic.actions.endQuery(queryId, ViewType.FUNNELS, result.last_refresh)
-                    actions.setSteps(result.result as FunnelStep[])
+                    if (params.breakdown && values.clickhouseFeatures) {
+                        const aggregatedResult = aggregateBreakdownResult(result)
+                        actions.setSteps(aggregatedResult.result)
+                        return aggregatedResult.result
+                    }
```

Comment:
I think we're doing things in a wrong order here. 

Ideally, a `loader` should just have one job: loading data from an API endpoint... or two. Here the loader also takes on the role of transforming this data and doing various other things.

If you want to transform loaded data into some other format, you should use a selector or a combination of selectors. E.g. `results` -> `filteredResults` -> `sortedResults`. 

Could some similar pattern work here as well? E.g. you just use a selector on `steps` that aggregates them? Then all places where you need this breakdown data just read from that selector.

Could this work?

Reply:
Yep, selectors are definitely the way to go. What's funky as well is that the second call to `pollFunnel` to fetch histogram bins data does not return to the enclosing scope, so that probably belongs in its own listener/loader.